youtube_dl/extractor/picarto.py

```diff
@@ -0,0 +1,91 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, js_to_json, urlencode_postdata
+
+
+class PicartoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?picarto\.tv/(?P<id>[a-zA-Z0-9]+)[^/]*$'
+    _TEST = {
+        'url': 'https://picarto.tv/Setz',
+        'info_dict': {
+            'id': 'Setz',
+            'ext': 'mp4',
+            'title': 're:^Setz [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
+            'timestamp': int,
+            'is_live': True
+        },
+        'params': {
+            'skip_download': True
+        }
+    }
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+        stream_page = self._download_webpage(url, channel_id)
+
+        if 'This channel does not exist.' in stream_page:
+            raise ExtractorError('Channel does not exist', expected=True)
+
+        player_settings_js = self._html_search_regex(
+            r'(?s)playerSettings\[1\] = (\{.+?\}\n)', stream_page,
```

Comment:
Again: relax regex.

Reply:
Could you be specific? Because the json capturing part is exactly as it needs to be and there can be  multiple playerSettings but we always want the first.

Reply:
It should accept any combination of whitespace.